statsd/statsd_test.go

```diff
@@ -198,6 +198,136 @@ func TestBufferedClient(t *testing.T) {
 
 }
 
+func TestJoinMaxSize(t *testing.T) {
```

Comment:
I think i did bring in your latest changes, had some conflicts because you squashed, but your changes should be here